hotosm/tasking-manager
# OSM Teams integration

This PR allows users to get members and managers from an OSM Teams team.

How to test:
- Create a new OAuth2 app on https://auth.mapping.team and copy the `CLIENT_ID` and `CLIENT_SECRET`. Use it to configure your TM instance, setting the following environment variables:
```
OSM_TEAMS_API_URL='https://dev.mapping.team' # it should use the dev instance now
OSM_TEAMS_CLIENT_ID=foo
OSM_TEAMS_CLIENT_SECRET=foo
```

- Go to the team management page, login to OSM Teams, choose an OSM Teams team to sync from and create a new team.

## Some screenshots
### Team page
<img width="468" alt="image" src="https://user-images.githubusercontent.com/666291/217862337-cc1eea8b-0c5a-4df0-b4cb-0d38799fc7f0.png">

### Team selection modal
<img width="638" alt="image" src="https://user-images.githubusercontent.com/666291/217862646-5ba37cc3-04fd-4506-8505-23028bc8f28f.png">

### Selected team details
<img width="635" alt="image" src="https://user-images.githubusercontent.com/666291/217863606-ff4ae461-7df7-4a22-955c-b021050985df.png">

### Managers and members imported
<img width="461" alt="image" src="https://user-images.githubusercontent.com/666291/217862809-3f1ae0e5-6ef1-4f4b-9860-c8b2ee112889.png">

### Resync action button
<img width="463" alt="image" src="https://user-images.githubusercontent.com/666291/217864318-239d1690-b0a1-4a4d-87ec-7525b1462f43.png">
